whynotearth/BrowTricks
# Feature: Refactor MenuHeader into general Header component

**Note: This PR is not ready to be merged before #47**

# Description

This PR does not address any specific issues. It involves attempting to change the 'BaseMenuHeader', which had a fixed icon, icon action, and inflexible heading text, into a more generalized 'BaseHeader' component.

Previously, we only had the 'BaseMenuHeader' component which we could use in the customer views which all use the same hamburger menu with slide-out drawer action:

![image](https://user-images.githubusercontent.com/20192983/81492458-a7497380-924c-11ea-9d44-40c9a2e290d2.png)

However, the existing 'BaseMenuHeader' component could be more generalized to fit more kinds of headers with different/more icons, text, actions, styling, etc., e.g.:

![image](https://user-images.githubusercontent.com/20192983/81492218-0dcd9200-924b-11ea-8f58-df30253f9d23.png)

# Commits

* 5c0a161: Rename the 'BaseMenuDrawer' component to 'CustomerMenuDrawer' to reflect its inflexibility as it will be used specifically for customer views.

* cc45467: Make 'BaseMenuHeader' component into 'Header' component which is more modular with slots:
  * `icon` - leftmost icon, has click emitter

    ![image](https://user-images.githubusercontent.com/20192983/81511312-46ae4b00-92cd-11ea-8bac-b6d3080345b9.png)

  * `content` - center text or other ui element

    ![image](https://user-images.githubusercontent.com/20192983/81511302-37c79880-92cd-11ea-9cd8-0358c17c4b39.png)

  * `icon-primary` - 2nd icon or additional text, has click emitter

    ![image](https://user-images.githubusercontent.com/20192983/81511286-1ebee780-92cd-11ea-8092-fa5a9869bf4a.png)

  * `icon-secondary` - 3rd icon, has click emitter

    ![image](https://user-images.githubusercontent.com/20192983/81511275-04850980-92cd-11ea-97b8-3382fc815643.png)

  * `menu-drawer` - holding the menu drawer component that will slide in

* a7df9cf: Refactor customer views to use the new Header component

## Commits after reviews

* b223c96: Put new component name into 'BaseHeader' component

* ee23538: Make parent tag of icon slot into `a` tag instead of `div` for accessiblity

* 2aee21e: Replace icon import names for clarity, e.g. 'MenuIcon' -> 'IconMenu'

# How to Test

There are no visual nor functional changes in this PR. Visit the three views and observe that there are no visual nor functional changes to speak of.

1. [Customer View - Home](https://deploy-preview-48--browtricks.netlify.app//shop/test-tenant-slug)
2. [Customer View - Business Hours](https://deploy-preview-48--browtricks.netlify.app/shop/test-tenant-slug/business-hours)
3. [Customer View - Contact Form](https://deploy-preview-48--browtricks.netlify.app/shop/test-tenant-slug/contact)

(Links will take you to the pages in the deploy preview)
